modules/gcp/compute.go

```diff
@@ -7,146 +7,259 @@ import (
 	"testing"
 
 	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
 	"golang.org/x/oauth2/google"
-	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/compute/v1"
 )
 
-// GetPublicIPOfInstance gets the public IP address of the given Instance in the given region.
-func GetPublicIPOfInstance(t *testing.T, projectID string, zone string, instanceID string) string {
-	ip, err := GetPublicIPOfInstanceE(t, projectID, zone, instanceID)
+// Corresponds to a GCP Compute Instance (https://cloud.google.com/compute/docs/instances/)
+type Instance struct {
+	projectID string
+	*compute.Instance
+}
+
+// Corresponds to a GCP Image (https://cloud.google.com/compute/docs/images)
+type Image struct {
+	projectID string
+	*compute.Image
+}
+
+// Corresponds to a GCP Zonal Instance Group (https://cloud.google.com/compute/docs/instance-groups/)
+type ZonalInstanceGroup struct {
+	projectID string
+	*compute.InstanceGroup
+}
+
+// Corresponds to a GCP Regional Instance Group (https://cloud.google.com/compute/docs/instance-groups/)
+type RegionalInstanceGroup struct {
+	projectID string
+	*compute.InstanceGroup
+}
+
+// FetchInstance queries GCP to return an instance of the (GCP Compute) Instance type
+func FetchInstance(t *testing.T, projectID string, name string) *Instance {
```

Comment:
Please see the error handling docs: https://github.com/gruntwork-io/terratest#error-handling. Each test method should have a `fooE` flavor that returns all errors and a `foo` version that calls `t.Fatal` if any error is returned. This applies to all new methods in this PR.

Reply:
Ah, thanks for clarifying. I thought there was more discretion around this but it makes sense that these are recommended for all methods. Updated in 78e3ad59fd8561c0fafcf0ad1aa4e15682b7799e.